Python/Product/PythonTools/PythonTools/Environments/AddCondaEnvironmentControl.xaml

```diff
@@ -0,0 +1,405 @@
+                            <DataTrigger Binding="{Binding Path=CondaPreview.IsInProgress}" Value="True">
+                                <Setter TargetName="ProgressBar" Property="UIElement.Visibility" Value="Visible"/>
+                            </DataTrigger>
+                            <DataTrigger Binding="{Binding Path=CondaPreview.IsNoPackages}" Value="True">
+                                <Setter TargetName="NoPackagesMessage" Property="UIElement.Visibility" Value="Visible"/>
+                            </DataTrigger>
+                            <DataTrigger Binding="{Binding Path=CondaPreview.HasPreviewError}" Value="True">
+                                <Setter TargetName="ErrorMessage" Property="UIElement.Visibility" Value="Visible"/>
+                            </DataTrigger>
+                            <MultiDataTrigger>
+                                <MultiDataTrigger.Conditions>
+                                    <Condition Binding="{Binding Path=CondaPreview.IsEnvFile}" Value="True" />
+                                    <Condition Binding="{Binding Path=CondaPreview.EnvFileContents}" Value=""  />
+                                </MultiDataTrigger.Conditions>
+                                <Setter TargetName="EnvFileMissing" Property="UIElement.Visibility" Value="Visible" />
+                            </MultiDataTrigger>
+                            <DataTrigger Binding="{Binding Path=CondaPreview.IsEnvFile}" Value="True">
+                                <Setter TargetName="EnvFileMessage" Property="UIElement.Visibility" Value="Visible"/>
+                            </DataTrigger>
+                            <DataTrigger Binding="{Binding Path=CondaPreview.IsNoEnvFile}" Value="True">
+                                <Setter TargetName="EnvFileMissing" Property="UIElement.Visibility" Value="Visible"/>
+                            </DataTrigger>
+                            <DataTrigger Binding="{Binding Path=CondaPreview.IsPackages}" Value="True">
+                                <Setter TargetName="PackagesList" Property="UIElement.Visibility" Value="Visible"/>
+                            </DataTrigger>
+                        </DataTemplate.Triggers>
+                    </DataTemplate>
+                </ResourceDictionary>
+            </ResourceDictionary.MergedDictionaries>
+        </ResourceDictionary>
+    </UserControl.Resources>
+    <Grid Margin="0 7 0 0">
+        <Grid Visibility="{Binding Path=IsCondaMissing, Converter={StaticResource BoolToVisible}}">
+            <TextBlock
+                HorizontalAlignment="Center"
+                VerticalAlignment="Center"
+                Text="{x:Static common:Strings.CondaNotFoundError}"
+                TextWrapping="Wrap"/>
+        </Grid>
+        <Grid Visibility="{Binding Path=IsCondaMissing, Converter={StaticResource BoolToHidden}}">
+            <Grid.ColumnDefinitions>
+                <ColumnDefinition Width="514*"/>
```

Comment:
`514*/188*` - is it to preserve a proportion?

Reply:
correct